src/application/router/routes.ts

```diff
@@ -65,6 +67,24 @@ const routes: RouteRecordRaw[] = [
       id: String(route.params.id),
     }),
   },
+
+  {
+    name: 'join',
+    path: '/join/:id',
+    component: Join,
+    props: route => ({
+      id: String(route.params.id),
+    }),
+    beforeEnter: async (to, from, next) => {
+      const noteSettings = await useNoteSettings().joinNote(to.params.id as string);
+
+      if (noteSettings?.id != null) {
```

Comment:
you need to handle joining errors and show user what happened and what he should need to do

Reply:
currently when a user isn't authenticated, he's signed or using google oauth popu. To display message i think i need to keep track of state

Please line 147, useNoteSettings.ts file

Reply:
currently when a user isn't authenticated, he's signed or using google oauth popu. To display message i think i need to keep track of state

Please line 147, useNoteSettings.ts file

Reply:
> you need to handle joining errors and show user what happened and what he should need to do

currently when a user isn't authenticated, he's signed or using google oauth popu. To display message i think i need to keep track of state

Please line 147, useNoteSettings.ts file

Reply:
In that case, i don't think handling the error on route file would help me to display 'what the user needs to do' rather on join.vue file would. Since i can't track state as its happens